### PR TITLE
fix: residual space after section causes bad parsing

### PR DESCRIPTION
**Pull request by wraithgar**
Any character left on a line after a section should be ignored.

### CODE REVIEW

**Comment by wraithgar** [comment]
Rebase/recreation of #123 to let CI run.

File: lib/ini.js [hunk 1]
```diff
@@ -67,8 +67,8 @@ const decode = str => {
   const out = Object.create(null)
   let p = out
   let section = null
-  //          section     |key      = value
-  const re = /^\[([^\]]*)\]$|^([^=]+)(=(.*))?$/i
+  //          section         |key      = value
```
Comment (wraithgar) [inline lib/ini.js:70]:
```suggestion
  //          section          |key      = value
```

**Comment by Nautigsam** [comment]
> Rebase/recreation of #123 to let CI run.

Hi, I can rebase my branch if you prefer.

**Comment by wraithgar** [comment]
@Nautigsam oh by all means! Most of the PRs were old enough that I didn't want to hassle the authors about it anymore.

Please note the `\s` change to the regex.

**Pull request closed without merging**